components/brave_search/renderer/brave_search_render_frame_observer.cc

```diff
@@ -0,0 +1,68 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_search/renderer/brave_search_render_frame_observer.h"
+
+#include <string>
+#include <vector>
+
+#include "base/no_destructor.h"
+#include "content/public/renderer/render_frame.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+
+namespace {
+
+static base::NoDestructor<std::vector<std::string>> g_vetted_hosts(
+    {"search.brave.com", "search-dev.brave.com"});
+
+bool IsVettedHost(const GURL& url) {
+  std::string host = url.host();
+  for (size_t i = 0; i < g_vetted_hosts->size(); i++) {
+    if ((*g_vetted_hosts)[i] == host)
+      return true;
+  }
+
+  return false;
+}
+
+}  // namespace
+
+namespace brave_search {
+
+BraveSearchRenderFrameObserver::BraveSearchRenderFrameObserver(
+    content::RenderFrame* render_frame)
+    : RenderFrameObserver(render_frame) {
+  native_javascript_handle_.reset(new BraveSearchJSHandler(render_frame));
+}
+
+BraveSearchRenderFrameObserver::~BraveSearchRenderFrameObserver() {}
+
+void BraveSearchRenderFrameObserver::DidStartNavigation(
+    const GURL& url,
+    base::Optional<blink::WebNavigationType> navigation_type) {
+  url_ = url;
```

Comment:
I'm not sure that we should get the url here since this isn't a committed navigation yet. It seems safer to just check against `render_frame()->GetWebFrame()->GetSecurityOrigin()` in `DidCreateScriptContext`